xayn-ai/src/coi/relevance.rs

```diff
@@ -144,6 +145,7 @@ impl RelevanceMap {
         self.coi_to_relevance
             .remove(&coi_id)
             .map(|relevances| {
+                self.cleaned.remove(&coi_id);
```

Comment:
removing the id from the `cleaned` might starve this map soon. the main issue is that `RelevanceMap::remove()` is called in every iteration of the feedback loop (via `RelevanceMap::select_key_phrases()`), but while the key phrases of the corresponding id are removed & unified with new candidates & selected & inserted again in the `coi_to_relevance`/`relevance_to_key_phrase` maps, this is not the case yet for the `cleaned` map which will starve it. on the other hand, if we ran another iteration of the feedback loop, than we have new key phrases and don't have to worry about the `cleaned` anymore, is this the idea?

Reply:
In the feedback loop we are removing the coi and put it back with updated data, so the coi is not clean anymore but we have fresh data in and the keywords that we just put in can be selected. 
After the remove the information about that coi in `cleaned` are stale.